test/e2e/virtual/apiexport/virtualworkspace_test.go

```diff
@@ -0,0 +1,227 @@
+
+	dynamicClients, err := dynamic.NewClusterForConfig(cfg)
+	require.NoError(t, err, "failed to construct dynamic cluster client for server")
+
+	kubeClusterClient, err := kubernetes.NewClusterForConfig(cfg)
+	require.NoError(t, err, "failed to construct kube cluster client for server")
+
+	wildwestClusterClient, err := wildwestclientset.NewClusterForConfig(cfg)
+	require.NoError(t, err, "failed to construct wildwest cluster client for server")
+
+	framework.AdmitWorkspaceAccess(t, ctx, kubeClusterClient, orgClusterName, []string{"user-1"}, nil, []string{"member"})
+	framework.AdmitWorkspaceAccess(t, ctx, kubeClusterClient, serviceProviderWorkspace, []string{"user-1"}, nil, []string{"member", "access"})
+	//Create provider
+	setUpServiceProvider(ctx, dynamicClients, kcpClients, kubeClusterClient, serviceProviderWorkspace, t)
+	// Need to Create a consumer
+	bindConsumerToProvider(ctx, consumerWorkspace, serviceProviderWorkspace, t, kcpClients)
+	// Need to create a cowby in Consumer
+	createCowboyInConsumer(ctx, t, consumerWorkspace, wildwestClusterClient)
+
+	t.Logf("test that the admin user can use the virtual workspace to get cowboys")
+	vc := virtualClusterClient{cfg}
+	wildwestVCClient, err := vc.Cluster(serviceProviderWorkspace, "today-cowboys")
+	require.NoError(t, err)
+	cowboysProjected, err := wildwestVCClient.Cluster(logicalcluster.Wildcard).WildwestV1alpha1().Cowboys("").List(ctx, metav1.ListOptions{})
+	require.NoError(t, err)
+	require.Equal(t, 1, len(cowboysProjected.Items))
+
+	// Attempt to use VW using user-1 should expect an error
+	t.Logf("Make sure that user-1 is denied")
+	user1VC := virtualClusterClient{config: userConfig("user-1", cfg)}
+	wwUser1VC, err := user1VC.Cluster(serviceProviderWorkspace, "today-cowboys")
+	require.NoError(t, err)
+	_, err = wwUser1VC.Cluster(logicalcluster.Wildcard).WildwestV1alpha1().Cowboys("").List(ctx, metav1.ListOptions{})
+	require.Error(t, err)
+
+	// Create clusterRoleBindings for content access.
+	t.Logf("create the cluster role and bindings to give access to the virtual worksapce")
+	cr, crb := createClusterRoleAndBindings("user-1-vw", "user-1", "User", []string{"list", "get"})
+	_, err = kubeClusterClient.Cluster(serviceProviderWorkspace).RbacV1().ClusterRoles().Create(ctx, cr, metav1.CreateOptions{})
+	require.NoError(t, err)
+	_, err = kubeClusterClient.Cluster(serviceProviderWorkspace).RbacV1().ClusterRoleBindings().Create(ctx, crb, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	// Get cowboys from the virtual workspace with user-1.
+	t.Logf("Get Cowboys with user-1")
+	require.Eventually(t, func() bool {
+		cb, err := wwUser1VC.Cluster(logicalcluster.Wildcard).WildwestV1alpha1().Cowboys("").List(ctx, metav1.ListOptions{})
+		if err != nil {
+			return false
+		}
+		require.Equal(t, 1, len(cb.Items))
+		return true
+	}, wait.ForeverTestTimeout, time.Millisecond*100, "expected user-1 to list cowboys from virtual workspace")
+
```

Comment:
would like to see a case where a user or service account gets read-only access.

Reply:
Added a test that checks that a user can only have `List` and not allowed to update